packages/coil-extension/src/background/services/BackgroundScript.ts

```diff
@@ -175,6 +176,53 @@ export class BackgroundScript {
     })
   }
 
+  private modifyPaymentHeader(details: any) {
+    // Check if the header has been defined already
+
+    let found = false
+    const paymentHeader = 'accept-payment'
+    const paymentPreference = 'webmon/*;q=.8'
+
+    // it's possible that another extension will be adding this header
+    // maybe an adblocker wants to add ads/*;q=0,ads/acceptable;q=.5
+    for (const header of details.requestHeaders) {
```

Comment:
In the ideal world multiple extensions would use this header. This looks to see if the header is already there, and if so it appends its payment preference. If not it creates.